altair/utils/core.py

```diff
@@ -785,68 +912,18 @@ def infer_encoding_types(args: Sequence, kwargs: MutableMapping, channels: Modul
         All args and kwargs in a single dict, with keys and types
         based on the channels mapping.
     """
-    # Construct a dictionary of channel type to encoding name
-    # TODO: cache this somehow?
-    channel_objs = (getattr(channels, name) for name in dir(channels))
-    channel_objs = (
-        c for c in channel_objs if isinstance(c, type) and issubclass(c, SchemaBase)
+    cache = (
```

Comment:
From my understanding, the only instance where `infer_encoding_types` is ever called (excluding tests) is within the `.encode` method.

In this PR, I have updated that call to not pass a `channels` argument https://github.com/dangotbanned/altair/blob/e4ab7052e9a1e62ff1fd80379864489c69a1e020/tools/generate_schema_wrapper.py#L258

> Or did you decide on making channels optional so that it's easier to cache as you then no longer call it with channels passed but to maintain backwards compatibility, if someone passes from_channels it does not use the cached version but instead determines it based on the provided channels

To simplify all of this, there are two cases:

1. Calling without `channels` == default == caching of `channels.py`
2. Calling with some module == no caching == identicial behaviour as before

I suppose it would be possible to cache any module passed in, but I think it would add a lot of new complexities.
- What guarantees can me make about other modules?
- Would this be scoped to modules in `altair`, or third-party?
- Will the cached metadata remain correct?

Reply:
Agree, ok for me to not cache any other modules as this was never the intended goal and this allows us to still have backwards compatibility in case anyone has ever used this function with another module (which would be unexpected)